openreview/venue/venue.py

```diff
@@ -1113,37 +1116,41 @@ def ithenticate_create_and_upload_submission(self):
             )
             iThenticate_submission_id = res["id"]
 
-            # Create a new edge with head as the note ID and tail as the iThenticate submission ID
-            iThenticate_edge = openreview.api.Edge(
-                invitation=self.get_iThenticate_plagiarism_check_invitation_id(),
-                head=submission.id,
-                tail=iThenticate_submission_id,
-                label="Created",
-                weight=-1,
-            )
+            if (self.client.get_edges_count(head=submission.id, invitation=self.get_iThenticate_plagiarism_check_invitation_id()) == 0):
+                iThenticate_edge = openreview.api.Edge(
+                    invitation=self.get_iThenticate_plagiarism_check_invitation_id(),
+                    head=submission.id,
+                    tail=iThenticate_submission_id,
+                    label="Created",
+                    weight=-1,
+                )
+
+            
+            else:
+                iThenticate_edge = self.client.get_edges(head=submission.id, invitation=self.get_iThenticate_plagiarism_check_invitation_id())[0]
```

Comment:
why do you need to do this?

Reply:
In the if statement before I checked if there was a pre-existing edge. If there exists an edge that means it was an edge with the label `'Error...'` So I retrieved the edge, reset its fields, and set the tail as the new submission ID.

Reply:
My only concern here is that if Error is coming from a later stage, for example: computing the report. We don't need to delete the submission and upload it again, we might need to retry recomputing the report.

Maybe we could have a flag that resubmit the edge with errors if it is on but by default I would ignore them.

Reply:
I think we can create submissions for notes that don't have an edge already, and we can have a separate function that handles edges with errors. In the function that handles edges with errors, 

1. We can make a call to `get_submission_info`, this will tell us if there was an error while uploading. 
2. We can repeat the upload request if an error is returned by `get_submission_info`. 
3. We submit a report generation request regardless of whether `get_submission_info` returned an error.

Reply:
Sounds good to me.

Reply:
I've added the function for error handling. I deviated slightly from this since it was difficult to determine how long I should wait to submit a similarity report request after uploading the file.
Now the steps look like this:
1. We make a call to `get_submission_info`, this will tell us if there was an error while uploading.
2. We repeat the upload request if an error is returned by `get_submission_info`.

Do steps 3 and 4 only if `get_submission_info` didn't return an error.

3. We make a call to `get_similarity_report_status`, this will tell us if there was an error while submitting the similarity report request.
4. We submit a report generation request if an error is returned by `get_similarity_report_status`.

We can use the `ithenticate_request_similarity_report` function to request the similarity report for cases where we re-uploaded the file.